src/urlencoded.rs

```diff
@@ -1,20 +1,19 @@
 use groupable::Groupable;
 use hyper::uri::RequestUri;
 use hyper::uri::RequestUri::{Star, AbsoluteUri, AbsolutePath, Authority};
-use plugin::{Plugin, Pluggable};
-use request::Request;
-use status::StatusCode;
 use std::collections::HashMap;
-use std::io::{Error, Read};
-use typemap::Key;
 use url::{form_urlencoded, UrlParser};
 
-type QueryStore = HashMap<String, Vec<String>>;
+pub type QueryStore = HashMap<String, Vec<String>>;
 
 #[derive(Debug, PartialEq, Eq)]
 pub struct Query(QueryStore);
 
 impl Query {
+    pub fn new(store: QueryStore) -> Query {
```

Comment:
Could probably just change the return type of `parse` as it seems to be the only usage?